examples/compare_connectivity_over_time_over_trial.py

```diff
@@ -0,0 +1,401 @@
+                             n_freq_bands, n_times))
+con_epochs_array[con_epochs_array == 0] = np.nan  # nan matrix
+
+# Compute connecitivty over trials
+con_epochs = spectral_connectivity_epochs(data_epoch,
+                                          method=connectivity_methods,
+                                          sfreq=sfreq, mode="cwt_morlet",
+                                          cwt_freqs=freqs, fmin=fmin,
+                                          fmax=fmax, faverage=True)
+
+# Get data as connectivity matrices
+for c in range(n_con_methods):
+    con_epochs_array[c] = con_epochs[c].get_data(output="dense")
+
+con_epochs_array = np.mean(con_epochs_array, axis=4)  # average over timepoints
+
+foi = list(Freq_Bands.keys()).index("alpha")  # frequency of interest
+
+fig, ax = plt.subplots(1, n_con_methods, figsize=(6 * n_con_methods, 6))
+for c in range(n_con_methods):
+    ax[c].imshow(con_epochs_array[c, :, :, foi], cmap="binary", vmin=0, vmax=1)
+    ax[c].set_title(connectivity_methods[c])
+    print(f"Connectivity method: {connectivity_methods[c]}\n" +
+          f"{con_epochs_array[c,:,:,foi]}")
+
+###############################################################################
+# We see that connectivity over trials are not 1, since the phase differences
+# between two channels are not the same over trials.
+#
+# We will now compute connectivity over time.
+
+# Pre-allocatate memory for the connectivity matrices
+con_time_array = np.zeros((n_con_methods, n_epochs,
+                           n_channels, n_channels, n_freq_bands))
+con_time_array[con_time_array == 0] = np.nan  # nan matrix
+
+con_time = spectral_connectivity_time(data_epoch, freqs,
+                                      method=connectivity_methods,
+                                      sfreq=sfreq, fmin=fmin, fmax=fmax,
+                                      faverage=True)
+
+# Get data as connectivity matrices
+for c in range(n_con_methods):
+    con_time_array[c] = con_time[c].get_data(output="dense")
+
+con_time_array = np.mean(con_time_array, axis=1)  # average over epochs
+foi = list(Freq_Bands.keys()).index("alpha")  # frequency of interest
+
+fig, ax = plt.subplots(1, n_con_methods, figsize=(6 * n_con_methods, 6))
+for c in range(n_con_methods):
+    ax[c].imshow(con_time_array[c, :, :, foi], cmap="binary", vmin=0, vmax=1)
+    ax[c].set_title(connectivity_methods[c])
+    print(f"Connectivity method: {connectivity_methods[c]}\n" +
+          f"{con_time_array[c,:,:,foi]}")
+
+###############################################################################
+# We see that for case 2, the connectivity over time is approximately 1,
+# since the phase differences over time between two channels are synchronized.
+
+###############################################################################
+# Real data demonstration
+# -----------------------
+# To finish this example, we will compute connectivity for a sample EEG data.
```

Comment:
Can you add some details on what we expect to see and the motivation for picking different analysis options?

E.g. we focus on theta power. Why? What do we expect to see? At the timepoint of highest-connectivity, what does that tell us?

Reply:
I haven't worked with ERP analysis myself, so I'm not really sure what to expect.
I talked with a colleague and she said you usually expect some theta power differences in the occipital cortex for a visual task, but wasn't sure regarding the expected connectivity, so I purposefully left it blank since I'm also not confident in the "ground truth" in the example.

Reply:
Perhaps what we can say is something along those lines (ideally with a reference perhaps)?

E.g.

```
Since the sample data is a visual task, we should generally expect to see some theta power
ERP(?) differences in the occipital cortex :footcite:`<insert_ref see references.bib and other files>`.
Here, we will therefore analyze the correlations in the frequency domain, centered at the theta band.
```

Reply:
I found this [reference](https://www.sciencedirect.com/science/article/pii/S0926641004000308?via%3Dihub) about how phase locked alpha and theta oscillations are related to P1 and N1 and tried to relate it to the results in the example.